worker/credentialvalidator/worker.go

```diff
@@ -67,20 +63,34 @@ func NewWorker(config Config) (worker.Worker, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	w, err := config.Facade.WatchCredential(mc.CloudCredential)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
+
+	// It is possible that this model is on the cloud that does not
+	// necessarily require authentication. Consequently, a model
+	// is created without a set credential. However, later on,
+	// a credential may be added to a model. This worker will need to
+	// be restarted as there will be no credential watcher set to react to
```

Comment:
Another option is to always start the watcher and have it react to any credential change. But I think your idea to bounce the worker is better / more efficient.